pycose/keys/cosekey.py

```diff
@@ -94,6 +94,25 @@ def from_dict(cls, received: dict) -> 'CK':
 
         return key_obj
 
+    @classmethod
+    def from_cryptography_key(
+        cls,
+        ext_key,
+        optional_params: Optional[dict] = None
+    ) -> 'CK':
```

Comment:
The uncontrained `CK` doesn't make much sense. It lets the caller choose an arbitrary subclass of `CoseKey`, and it also breaks type inference.

It would make something like `key: EC2Key = CoseKey.from_cryptography_key(...)` typecheck regardless of the value of the arguments, and on the other hand, `key = CoseKey.from_cryptography_key(...)` would fail.

This should just be `def from_cryptography_key(cls, ...) -> "CoseKey"`.

An alternative is to contrain `CK` by using it as the type of `cls` using `cls: Type['CK']`, as I did in https://github.com/TimothyClaeys/pycose/pull/102. In this case however, it is crucial that this method have a check before returning to make sure the return value is an instance of `cls` (otherwise eg. `SymmetricKey.from_cryptography_key(rsa_obj)` would have type SymmetricKey, but return an instance of an RSAKey).

(Of course this applies to the existing `from_dict` and `decode` methods as well, feel free to fix them or not. Actually most/all uses of typevars in this file seems either unnecessary or wrong)